cmd/frontend/graphqlbackend/git_commit.go

```
@@ -417,3 +475,27 @@ func (r *GitCommitResolver) canonicalRepoRevURL() *url.URL {
 	repoUrl.Path += "@" + string(r.oid)
 	return &repoUrl
 }
+
+var p4FusionCommitSubjectPattern = regexp.MustCompile(`^(\d+) - (.*)$`)
+
+func parseP4FusionCommitSubject(subject string) (string, error) {
+	matches := p4FusionCommitSubjectPattern.FindStringSubmatch(subject)
+	if len(matches) != 3 {
+		return "", errors.Newf("failed to parse commit subject %q for commit that was converted by p4-fusion", subject)
+	}
+	return matches[2], nil
+}
+
+// Either git-p4 or p4-fusion could be used to convert a perforce depot to a git repo. In which case the
+// [git-p4: depot-paths = "//test-perms/": change = 83725]
+// [p4-fusion: depot-paths = "//test-perms/": change = 80972]
+var gitP4Pattern = regexp.MustCompile(`\[(?:git-p4|p4-fusion): depot-paths = "(.*?)"\: change = (\d+)\]`)
+
+func getP4ChangelistID(input string) string {
```

I think returning `(string, error)` (and not a default zero-value in case of an error) is a more go-way approach and also uses the same code style which you used in `parseP4FusionCommitSubject`.

---

Hmm, I don't have a strong opinion here about which way to go. I didn't want to return a new error since we don't really have a real use case of that error as it will need to be ignored and saw that as adding unnecessary complexity to the call stack.

Umm, maybe it's worth an error + logging it and then going ahead with existing behaviour when I think about it. 🤔 

Still thinking on this...

---

Fixed.